tests/02_eradiate/01_unit/scenes/atmosphere/test_heterogeneous.py

```diff
@@ -306,11 +313,92 @@ def test_heterogeneous_blend_switches(mode_mono):
+):
+    """
+    Phase function weights are correct when the molecular atmosphere is
+    absorbing-only and the particle layer is either absorbing-only or
+    scattering-only.
+    """
+    # Create the heterogeneous atmosphere
+    pl_bottom = 1.0 * ureg.km  # arbitrary
+    pl_top = 4.0 * ureg.km  # arbitrary
+    particle_layer = ParticleLayer(
+        bottom=pl_bottom,
+        top=pl_top,
+        dataset=scattering_only,
+    )
+    atmosphere1 = HeterogeneousAtmosphere(
+        molecular_atmosphere=MolecularAtmosphere.afgl_1986(
+            has_absorption=True,
+            has_scattering=False,
+        ),
+        particle_layers=particle_layer,
+        geometry={"type": "spherical_shell"},
+    )
+    atmosphere2 = attrs.evolve(
+        atmosphere1,
+        particle_layers=ParticleLayer(
+            bottom=pl_bottom,
+            top=pl_top,
+            dataset=absorbing_only,
+        ),
+    )
+
+    # Generate kernel dict
+    spectral_ctx = CKDSpectralContext(bin_set="1nm")
+    ctx = KernelDictContext(
+        spectral_ctx=spectral_ctx,
+    )
+    kernel_dict1 = atmosphere1.kernel_phase(ctx)
+    kernel_dict2 = atmosphere2.kernel_phase(ctx)
+
+    # Write phase function weight data to disk
+    weight_data1 = kernel_dict1.data["phase"]["weight"]["grid"]
+    path1 = tmp_path / "gridvolume1.vol"
+    weight_data1.write(str(path1))
+    weight_data2 = kernel_dict2.data["phase"]["weight"]["grid"]
+    path2 = tmp_path / "gridvolume2.vol"
+    weight_data2.write(str(path2))
+
+    # Read the phase function weight data from disk
+    weight1 = read_binary_grid3d(path1)
+    z_units = "km"
+    weight1 = xr.DataArray(
+        weight1.squeeze(),
+        dims=["z"],
+        coords={
+            "z": (
+                "z",
+                atmosphere1._high_res_z_layer().m_as(z_units),
+                {"units": z_units},
             ),
-            particle_layers=[ParticleLayer()],
-        )
+        },
+    )
+    weight2 = read_binary_grid3d(path2)
+    weight2 = xr.DataArray(
+        weight2.squeeze(),
+        dims=["z"],
+        coords={
+            "z": (
+                "z",
+                atmosphere2._high_res_z_layer().m_as(z_units),
+                {"units": z_units},
+            ),
+        },
+    )
+
+    # within the particle layer, the phase function weight should be
+    # * > 0, if the particle layer medium is scattering
+    # * else 0
+    inside_particle_layer = (weight1.z >= pl_bottom.m_as(z_units)) & (
+        weight1.z <= pl_top.m_as(z_units)
+    )
+    assert np.all(weight1.where(inside_particle_layer, drop=True).values > 0.0)
+    assert np.all(weight2.where(inside_particle_layer, drop=True).values == 0.0)
+
+    # outside the particle layer, the phase function weight should be zero.
+    assert np.all(weight1.where(~inside_particle_layer, drop=True).values == 0.0)
+    assert np.all(weight2.where(~inside_particle_layer, drop=True).values == 0.0)
```

Comment:
Since you know the kernel units and the mesh, you can formulate this without masks or units. To simplify the test, you could use a particle layer covering exactly half the height of the atmosphere (it's much easier to cut it in two pieces and say "the bottom is in the particle layer").